src/custom/components/SearchModal/Manage/ManageMod.tsx

```diff
@@ -0,0 +1,98 @@
+import { Trans } from '@lingui/macro'
+import React, { useState } from 'react'
+import { PaddedColumn, Separator } from '@src/components/SearchModal/styleds'
+import { RowBetween } from 'components/Row'
+import { ArrowLeft } from 'react-feather'
+import { Text } from 'rebass'
+import { CloseIcon } from 'theme'
+import styled from 'styled-components/macro'
+// import { Token } from '@uniswap/sdk-core'
+import { ManageLists } from 'components/SearchModal/ManageLists'
+import ManageTokens from 'components/SearchModal/ManageTokens'
+// import { TokenList } from '@uniswap/token-lists'
+import { CurrencyModalView } from '@src/components/SearchModal/CurrencySearchModal'
+import { ManageProps } from './index' // mod
+
+const Wrapper = styled.div`
+  width: 100%;
+  position: relative;
+  padding-bottom: 80px;
+`
+
+// const ToggleWrapper = styled(RowBetween)`
+//   background-color: ${({ theme }) => theme.bg3};
+//   border-radius: 12px;
+//   padding: 6px;
+// `
+
+// const ToggleOption = styled.div<{ active?: boolean }>`
+//   width: 48%;
+//   padding: 10px;
+//   display: flex;
+//   align-items: center;
+//   justify-content: center;
+//   border-radius: 12px;
+//   font-weight: 600;
+//   background-color: ${({ theme, active }) => (active ? theme.bg1 : theme.bg3)};
+//   color: ${({ theme, active }) => (active ? theme.text1 : theme.text2)};
+//   user-select: none;
+
+//   :hover {
+//     cursor: pointer;
+//     opacity: 0.7;
+//   }
+// `
+
+export default function Manage({
+  onDismiss,
+  setModalView,
+  setImportList,
+  setImportToken,
+  setListUrl,
+  ToggleOption, // mod
+  ToggleWrapper, // mod
+}: ManageProps) {
+  // mod
+  // {
+  //   onDismiss: () => void
+  //   setModalView: (view: CurrencyModalView) => void
+  //   setImportToken: (token: Token) => void
+  //   setImportList: (list: TokenList) => void
+  //   setListUrl: (url: string) => void
+  // }
+  // toggle between tokens and lists
+  const [showLists, setShowLists] = useState(true)
+
+  return (
+    <Wrapper>
+      <PaddedColumn>
+        <RowBetween>
+          <ArrowLeft style={{ cursor: 'pointer' }} onClick={() => setModalView(CurrencyModalView.search)} />
+          <Text fontWeight={500} fontSize={20}>
+            <Trans>Manage</Trans>
+          </Text>
+          <CloseIcon onClick={onDismiss} />
+        </RowBetween>
+      </PaddedColumn>
+      <Separator />
+      <PaddedColumn style={{ paddingBottom: 0 }}>
+        {ToggleWrapper && // mod
```

Comment:
Optional type, thus doing checks here. This because the Type it's inferred from is shared with another file (which doesn't have these extra custom prop types).

Reply:
So for this one if we should always expect there to be a ToggleWrapper and ToggleShower you can change the type in the _index_ file to always require both and then to the default export give it a new type that is a variation of ManageProps but using Omit generic type. So in the index make it like Omit<ManageProps, 'ToggleShower' | 'ToggleWrapper'> and you don't need the conditional render after

Reply:
@W3stside Actually that's what I did initially (looked at previous code committed by use as an example). But this file `src/components/SearchModal/CurrencySearchModal.tsx` then complains:

`Property 'ToggleOption' is missing in type '{ onDismiss: () => void; setModalView: Dispatch<SetStateAction<CurrencyModalView>>; setImportToken: Dispatch<SetStateAction<Token | undefined>>; setImportList: Dispatch<...>; setListUrl: Dispatch<...>; }' but required in type 'Omit<ManageProps, "ToggleShower" | "ToggleWrapper">'.`

Because that file import our modded `Manage` component. What's the best way to solve for that, in this case?

Reply:
ah, I see. Makes sense from the surface. Let me take a quick look but in that case you're likely right

Reply:
fixed in #1035